src/lib/Simplify.hs

```diff
@@ -490,6 +490,7 @@ simplifyAtom atom = confuseGHC >>= \_ -> case atom of
   Con con -> Con <$> (inline traversePrimCon) simplifyAtom con
   TC tc -> TC <$> (inline traversePrimTC) simplifyAtom tc
   Eff eff -> Eff <$> substM eff
+  EffOp eff idx ty -> EffOp <$> substM eff <*> pure idx <*> substM ty  -- TODO(alex): check correctness
```

Comment:
This might be a good place for an `error "unsupported"` :)

Reply:
EffOp removed in #1070